system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -28,34 +30,50 @@ namespace system_metrics_collector
 
 PeriodicMeasurementNode::PeriodicMeasurementNode(
   const std::string & name,
-  const std::chrono::milliseconds measurement_period,
-  const std::string & publishing_topic,
-  const std::chrono::milliseconds publish_period)
-: Node(name),
-  measurement_period_(measurement_period),
-  publishing_topic_(publishing_topic),
-  measurement_timer_(nullptr),
-  publish_timer_(nullptr),
-  publish_period_(publish_period)
+  const rclcpp::NodeOptions & options)
+: Node(name, options)
 {
+  rcl_interfaces::msg::IntegerRange positive_range;
+  positive_range.from_value = 1;
+  positive_range.to_value =
+    std::numeric_limits<decltype(rcl_interfaces::msg::IntegerRange::to_value)>::max();
+  positive_range.step = 1;
+
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+  descriptor.read_only = true;
+  descriptor.integer_range.push_back(positive_range);
+
   // rclcpp::Node throws if name is empty
 
-  if (measurement_period <= std::chrono::milliseconds{0}) {
+  descriptor.description = "The period in milliseconds between each measurement";
+  auto measurement_period = declare_parameter(
+    collector_node_constants::kCollectPeriodParam,
+    collector_node_constants::kDefaultCollectPeriodInMs,
+    descriptor);
+  measurement_period_ = std::chrono::milliseconds{measurement_period};
+  if (measurement_period_ <= std::chrono::milliseconds{0}) {
     throw std::invalid_argument{"measurement_period cannot be negative"};
   }
 
-  if (publishing_topic.empty()) {
-    throw std::invalid_argument{"publishing_topic cannot be empty"};
-  }
-
-  if (publish_period <= std::chrono::milliseconds{0}) {
+  descriptor.description = "The period in milliseconds between each published MetricsMessage";
+  auto publish_period = declare_parameter(
+    collector_node_constants::kPublishPeriodParam,
+    collector_node_constants::kDefaultPublishPeriodInMs,
+    descriptor);
+  publish_period_ = std::chrono::milliseconds{publish_period};
+  if (publish_period_ <= std::chrono::milliseconds{0}) {
     throw std::invalid_argument{"publish_period cannot be negative"};
   }
```

Comment:
This is actually redundant now. `declare_parameter()` will throw before reaching this `if` statement. Thoughts on removing this check?

Reply:
+1

Reply:
Done.